File: src/MaksIT.CertsUI/Models/Identity/User/UserResponse.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.CertsUI.Models.Identity.User;

public class UserResponse : ResponseModelBase {
  #region Master data Properties
  public required Guid Id { get; set; }
  public required string Username { get; set; }
  public string? Email { get; set; }
  public string? MobileNumber { get; set; }
  public bool IsActive { get; set; }
  #endregion

  #region Two-factor authentication properties
  public bool TwoFactorEnabled { get; set; }
  public List<string>? TwoFactorRecoveryCodes { get; set; }
  public string? QrCodeUrl { get; set; }
  public int? RecoveryCodesLeft { get; set; }
  #endregion

  #region Authorization properties
  public bool IsGlobalAdmin { get; set; }
  public List<UserEntityScopeResponse> EntityScopes { get; set; } = [];
  #endregion
}

File: src/MaksIT.WebUI/src/models/user/UserResponse.ts
import type { ResponseModelBase } from '@maks-it.com/webui-contracts'
import { UserEntityScopeResponse } from './UserEntityScopeResponse'

/** Matches <c>MaksIT.CertsUI.Models.Identity.User.UserResponse</c> (camelCase JSON). */
export interface UserResponse extends ResponseModelBase {
  id: string
  username: string
  email?: string | null
  mobileNumber?: string | null
  isActive: boolean

  twoFactorEnabled?: boolean
  twoFactorRecoveryCodes?: string[]
  qrCodeUrl?: string
  recoveryCodesLeft?: number

  isGlobalAdmin?: boolean
  entityScopes?: UserEntityScopeResponse[]
}
